manifests/client/ssh.pp

```diff
@@ -5,34 +5,38 @@
 # Automatically included by profile_xcat::master
 class profile_xcat::client::ssh {
 
-    # Get xcat master node info
-    $pubkey = lookup( 'profile_xcat::master::root::sshkey_pub' )
-    $master_node_ip = lookup( 'profile_xcat::master_node_ip' )
+  # Get xcat master node info
+  $pubkey = lookup( 'profile_xcat::master::root::sshkey_pub' )
+  $master_node_ip = lookup( 'profile_xcat::master_node_ip' )
 
-    # Open firewall and configure sshd_config
-    $params = {
-        'PubkeyAuthentication'  => 'yes',
-        'PermitRootLogin'       => 'without-password',
-        'AuthenticationMethods' => 'publickey',
-        'Banner'                => 'none',
-    }
+  # Open firewall and configure sshd_config
+  $params = {
+    'PubkeyAuthentication'  => 'yes',
+    'PermitRootLogin'       => 'without-password',
+    'AuthenticationMethods' => 'publickey',
+    'Banner'                => 'none',
+  }
+  if $master_node_ip =~ String[1] {
     ::sshd::allow_from{ 'profile_xcat-client-ssh':
-        hostlist                => [ $master_node_ip ],
-        users                   => [ root ],
-        additional_match_params => $params,
+      hostlist                => [ $master_node_ip ],
+      users                   => [ root ],
+      additional_match_params => $params,
     }
+  }
 
-    # Authorize root's public key (from xcat master node)
+  # Authorize root's public key (from xcat master node)
+  if $pubkey =~ String[1] {
```

Comment:
What is this doing? Just checking to see if it is a String type?

Reply:
Yes, checks if pubkey is a String with length >= 1 (ie: fail for null string or any other type.)